### PR TITLE
Fix a reexport case for comment reference linking to extension methods

### LINKED ISSUE #3325: Links to extension methods do not work
I have an extension `CoreChecks` which is in scope, that extension defines a method `equals`. I hoped that a doc comment `/// [CoreChecks.equals]` would have a link to the docs for `equals`, but it renders as a code block instead of a link.

The analyzer jump to definition does work so this is a case where analyzer and dart doc disagree about comment references.

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #3325.  Also introduces a new test helper for reflective tests that should make it easier to test canonicalization edge cases.

### CODE REVIEW

File: test/dartdoc_test_base.dart [hunk 1]
```diff
@@ -11,6 +11,16 @@ import 'package:meta/meta.dart';
 import 'src/test_descriptor_utils.dart' as d;
 import 'src/utils.dart';
 
+/// Exception thrown for invalid use of [DartdocTestBase]'s api/
```
Comment (srawlins) [inline test/dartdoc_test_base.dart:14]: Should that end with a period?

**Review by srawlins — APPROVED** [review]
Very cool, cool test setup

File: test/dartdoc_test_base.dart (hunk 1 above)
Comment (jcollins-g) [reply test/dartdoc_test_base.dart]: fixed.
